### PR TITLE
fix: hot join issues for zombies & campaign

### PR DESCRIPTION
**Pull request by diamante0018**

### CODE REVIEW

File: src/client/component/bots.cpp [hunk 1]
```diff
@@ -94,9 +94,9 @@ namespace bots
 		}
 
 		int format_bot_string(char* buffer, [[maybe_unused]] const char* format, const char* name, const char* xuid,
-			const char* xnaddr, int protocol, int net_field_chk, const char* session_mode, int qport)
+		                      const char* xnaddr, int protocol, int net_field_chk, const char* session_mode, int qport)
 		{
-			const auto find_name = [](const std::string& needle) -> const char*
```
Comment (momo5502) [inline src/client/component/bots.cpp:99]: please separate refactorings and feature/fix pull requests. having all in one pull requests is bad

**Comment by momo5502** [comment]
what is the hot join issue?

File: src/client/component/patches.cpp [hunk 2]
```diff
@@ -2,64 +2,101 @@
...
+			{
+				game::Scr_AddInt(game::SCRIPTINSTANCE_SERVER, lobby_min_players->current.value.integer);
+			}
+
+			const auto num_expected_players = std::max(1, game::LobbyHost_GetClientCount(game::LOBBY_TYPE_GAME, game::LOBBY_CLIENT_TYPE_ALL));
+			game::Scr_AddInt(game::SCRIPTINSTANCE_SERVER, num_expected_players);
+		}
+	}
+
+	namespace mp
```
Comment (momo5502) [inline src/client/component/patches.cpp:43]: client

File: src/client/component/patches.cpp [hunk 3]
```diff
@@ -2,64 +2,101 @@
...
 				return;
 			}
 
-			sv_execute_client_messages_hook.invoke<void>(client, msg);
+			game::SV_ExecuteClientMessage(client, msg);
 		}
+	}
 
-		void script_errors_stub(const char* file, int line, unsigned int code, const char* fmt, ...)
+	namespace dedi
```
Comment (momo5502) [inline src/client/component/patches.cpp:58]: server

File: src/client/component/patches.cpp [hunk 4]
```diff
@@ -2,64 +2,101 @@
...
-				va_end(ap);
+				client->reliableAcknowledge = client->reliableSequence;
+				game::SV_DropClient(client, "EXE_LOSTRELIABLECOMMANDS", true, true);
+				return;
 			}
 
-			game::Com_Error(game::ERROR_SCRIPT_DROP, "%s", buffer);
+			game::SV_ExecuteClientMessage(client, msg);
 		}
 	}
```
Comment (momo5502) [inline src/client/component/patches.cpp:71]: there is no reason to duplicate the code here. the dedi function should be enough.

File: src/client/component/status.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		{
+			client_num_ = client_num;
+			game::Com_Printf(channel, label, fmt, client_num);
+		}
+
+		void print_client_xuid(int channel, int label, [[maybe_unused]] const char* fmt, const uint64_t xuid)
+		{
+			if (game::SV_IsTestClient(client_num_))
+			{
+				game::Com_Printf(channel, label, "%16s ", "bot0");
```
Comment (momo5502) [inline src/client/component/status.cpp:23]: what does this have to do with the PR?

**Review by momo5502 — CHANGES_REQUESTED** [review]
please split up your pr. it is not descriptive at all and contains a whole bunch of heterogenous changes.

File: src/client/component/status.cpp (hunk 5 above)
Comment (diamante0018) [reply src/client/component/status.cpp:23]: The right answer is nothing.
What it does is it prettifies the status command. Will try to break out of this PR.

**Comment by diamante0018** [comment]
removed status.cpp from this PR
It should be much more compact now and easier to review.

**Comment by diamante0018** [comment]
renamed to: `fix: hot join issues for zombies & campaign`
But this also fixes a dangling pointer in scripts.cpp

**Review by momo5502 — APPROVED** [review]
